tools/FlatFlowMigrationCli/MigrationLogger.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.DotNet.ProductConstructionService.Client.Models;
+using Microsoft.Extensions.Logging;
+
+namespace FlatFlowMigrationCli;
+
+/// <summary>
+/// Class that logs the operations in console and into a file instead of performing them.
+/// </summary>
+internal class MigrationLogger : ISubscriptionMigrator
+{
+    private readonly ILogger<MigrationLogger> _logger;
+    private readonly string _outputPath;
+
+    public MigrationLogger(ILogger<MigrationLogger> logger, string outputPath)
+    {
+        _logger = logger;
+        _outputPath = outputPath;
+    }
+
+    public async Task DisableSubscriptionAsync(Subscription subscription)
+    {
+        _logger.LogInformation("Would disable a subscription {sourceRepository} -> {targetRepository} / {subscriptionId}",
+            RemoveUrlPrefix(subscription.SourceRepository),
+            RemoveUrlPrefix(subscription.TargetRepository),
+            subscription.Id);
+
+        await LogActionAsync(GetActionKey(subscription), Action.Disable, subscription.Id.ToString());
+    }
+
+    public async Task DeleteSubscriptionAsync(Subscription subscription)
+    {
+        _logger.LogInformation("Would delete an existing subscription {sourceRepository} - {targetRepository} / {subscriptionId}...",
+            RemoveUrlPrefix(subscription.SourceRepository),
+            RemoveUrlPrefix(subscription.TargetRepository),
+            subscription.Id);
+        await LogActionAsync(GetActionKey(subscription), Action.Delete, subscription.Id.ToString());
+    }
+
+    public async Task CreateVmrSubscriptionAsync(Subscription subscription)
+    {
+        _logger.LogInformation("Would create subscription {vmrUri} - {repoUri}",
+            RemoveUrlPrefix(Constants.VmrUri),
+            RemoveUrlPrefix(subscription.TargetRepository));
+        await LogActionAsync($"{Constants.VmrUri} - {subscription.TargetRepository}", Action.Create, null, new()
```

Comment:
```suggestion
        await LogActionAsync($"{Constants.VmrUri} -> {subscription.TargetRepository}", Action.Create, null, new()
```

Reply:
THis one didn't work - the JSON key had a unicode sequency instead of `>`. So that was originally why I removed `->`. But we can keep it in the log statements